### PR TITLE
PP-366 (again): Import the latest Android build script and fix the build.

### PR DESCRIPTION
**Pull request by io7m**
**What's this do?**
This imports the latest Android build script and fix the build.

**Why are we doing this? (w/ JIRA link if applicable)**
https://ebce-lyrasis.atlassian.net/browse/PP-366

**How should this be tested? / Do these changes have associated tests?**
The CI is enough.

**Dependencies for merging? Releasing to production?**
None.

**Have you updated the changelog?**
N/A

**Has the application documentation been updated for these changes?**
N/A

**Did someone actually run this code to verify it works?**
@io7m ran the build and test suite.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -616,11 +694,101 @@ allprojects {
...
+        configurations.all {
+            configurationsActual.add(this.name)
+        }
+        File("configurations.txt").writeText(configurationsActual.joinToString("\n"))
+    }
+
+    afterEvaluate {
+        configurations.all {
+            isTransitive = transitiveConfigurations.contains(name)
+            // resolutionStrategy.failOnVersionConflict()
```
Comment (nunommts) [inline build.gradle.kts:770]: Was this comment left here on purpose?

**Review by nunommts — CHANGES_REQUESTED** [review]
Looks good! Just a really tiny probable issue I left on the review

File: build.gradle.kts (hunk 1 above)
Comment (io7m) [reply build.gradle.kts:770]: It was copied in from the `android-core` repository script. It's fine to remove it, but I left it in as I found myself repeatedly turning it on and off during development. It's something I'd like to leave permanently on, but we're not quite there yet.

Comment (nunommts) [reply build.gradle.kts:770]: Sure, there's no problem in keeping it. I was just asking

**Review by nunommts — APPROVED** [review]

**Pull request merged**